### PR TITLE
plugins/languages: move to by-name

### PR DESCRIPTION
**Pull request by khaneliman**
Also just individual plugins that don't share code actually.

### CODE REVIEW

**Comment by khaneliman** [comment]
Oops looks like I need to add stuff to the new ltex-extra test, real quick.

**Review by traxys — APPROVED** [review]

**Comment by khaneliman** [comment]
> I was hesitant to move `languages/*` given how "generic" some of the names will be... But I suppose since the name always matches `plugin.<name>` from the option path it shouldn't be an issue?
> 
> Is there any value in "language" plugins being grouped together in a category vs being in by-name?

I didn't think so, like you said, they are unique by-name already and it doesn't help me to find several of the plugins when they are nested that way.

File: tests/test-sources/plugins/by-name/ltex-extra/default.nix
Comment (MattSturgeon) [inline tests/test-sources/plugins/by-name/ltex-extra/default.nix:1]: Just double checking, this is intentionally a new file, not a half-done rename, right?

File: tests/test-sources/plugins/by-name/nix/default.nix
Comment (MattSturgeon) [inline tests/test-sources/plugins/by-name/nix/default.nix:1]: Same

File: tests/test-sources/plugins/by-name/plantuml-syntax/default.nix
Comment (MattSturgeon) [inline tests/test-sources/plugins/by-name/plantuml-syntax/default.nix:1]: Same

File: tests/test-sources/plugins/by-name/rust-tools/default.nix [hunk 1]
```diff
@@ -71,4 +71,13 @@
...
   };
+
+  rust-analyzer-options = {
+    plugins.rust-tools = {
+      enable = true;
+      server = {
+        assist.expressionFillDefault = "default";
+      };
+    };
+  };
```
Comment (MattSturgeon) [inline tests/test-sources/plugins/by-name/rust-tools/default.nix:82]: This feels unrelated to the PR? Is it testing a regression you encountered while moving files around?

**Comment by khaneliman** [comment]
@mergifyio queue

**Review by MattSturgeon — APPROVED** [review]
Everything looks good if you're happy with the files I flagged below :grin:

File: tests/test-sources/plugins/by-name/rust-tools/default.nix (hunk 1 above)
Comment (khaneliman) [reply tests/test-sources/plugins/by-name/rust-tools/default.nix:82]: This was just to make sure the imports were being tested since I saw a similar test in the rustaceanvim plugin.

File: tests/test-sources/plugins/by-name/ltex-extra/default.nix
Comment (khaneliman) [reply tests/test-sources/plugins/by-name/ltex-extra/default.nix:1]: There were a few plugins that didn't have tests

File: tests/test-sources/plugins/by-name/rust-tools/default.nix (hunk 1 above)
Comment (khaneliman) [reply tests/test-sources/plugins/by-name/rust-tools/default.nix:82]: Just added it since we were moving files around and wanted to double check the plugin could still access the out of directory options file.

File: tests/test-sources/plugins/by-name/ltex-extra/default.nix
Comment (MattSturgeon) [reply tests/test-sources/plugins/by-name/ltex-extra/default.nix:1]: Thought so, just wanted to double check. :grin:
